low/low.c

```diff
@@ -17,11 +17,34 @@
 
 // #define DEBUG
 // #define VERBOSE
+#define AVX
+
+#ifdef AVX
+#define mbufClear(buf) \
+	_mm256_storeu_ps((float*)(buf + mbufStructSize), zero256);
+#else
+#define mbufClear(buf) \
+	_mm_storeu_ps((float*)(buf + mbufStructSize), zero128); \
+	_mm_storeu_ps((float*)(buf + mbufStructSize + 16), zero128);
+#endif
+
+#define mbufInit(buf) \
+	mbufClear(buf) \
+	_mm_storeu_ps((float*)(buf + mbufStructSize + 32), zero128); \
+	*(char**)((char*)(buf) + mbufStructSize + 48) = (char*)(buf) + defaultStart; \
+	*(char**)((char*)(buf) + mbufStructSize + 56) = (char*)(buf);
 
-struct rte_mempool * mp;
 long receive_received = 0, receive_pushed = 0;
 long send_required = 0, send_sent = 0;
 long stop_freed = 0;
+
+int mbufStructSize;
+int headroomSize;
+int defaultStart;
+
+__m128 zero128;
+__m256 zero256;
+
```

Comment:
Add initialization with zeroes to these constants and a const modifier if compiler allows it for C code.